app/src/main/java/vn/sunasterisk/realfootball/data/remote/NetworkInterceptor.kt

```diff
@@ -0,0 +1,25 @@
+package vn.sunasterisk.realfootball.data.remote
+
+import okhttp3.Interceptor
+import okhttp3.Response
+import vn.sunasterisk.realfootball.FootballApplication
+import vn.sunasterisk.realfootball.R
+import vn.sunasterisk.realfootball.exceptions.NetworkException
+import vn.sunasterisk.realfootball.utils.NetworkUtil
+
+class NetworkInterceptor : Interceptor {
+
+    @Throws(NetworkException::class)
+    override fun intercept(chain: Interceptor.Chain): Response {
+        if (!NetworkUtil.isNetworkAvailable()) {
+            FootballApplication.applicationContext?.let {
+                throw NetworkException(it.getString(R.string.error_connect))
+            }
+        }
+        return chain.proceed(chain.request())
+    }
+
+    companion object {
+        fun newInstance() = NetworkInterceptor()
```

Comment:
nếu mà khởi tạo thông qua hàm như này thì nên để private constructor để tránh việc khởi tạo trực tiếp từ bên ngoài, như thế viết như thế mới có ý nghĩa
```suggestion
        class NetworkInterceptor private constructor() : Interceptor {
```